### PR TITLE
feat: Visually align Source in Editor view [env-switcher]

### LINKED ISSUE #283: [env-switcher] visually align "source" item with the others in the popover
I think there are some inconsistencies with the source item which could be confusing to users:

### Current state
**Editor**
The source item is missing here, which makes the env switcher look inconsistent
image

**Preview(/Live/Prod)**
The source item looks different than the others 
image

### Desired state
**Editor**
The source item should be added at the top with a gray background to indicate selected environment. We can also display the last modified date.
image

**Preview(/Live/Prod)**
image

### PR DESCRIPTION
**Pull request by asthabh23**
## Description

[env-switcher] visually align "source" item with the others in the popover (along with last modified/edited timestamp)

## Related Issue

Fix #283 

## Screenshots (if appropriate):
**Preview:**
image

**Editor:**
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/extension/app/components/plugin/env-switcher/env-switcher.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ export class EnvironmentSwitcher extends ConnectedElement {
    * @returns {string} - The last modified label
    */
   getLastModifiedLabel(id, lastModified) {
-    const envId = id === 'dev' ? 'preview' : id;
+    const envId = (id === 'dev') ? 'preview' : id;
```
Comment (rofe) [inline src/extension/app/components/plugin/env-switcher/env-switcher.js:117]: You can drop the brackets, too

**Review by rofe — APPROVED** [review]

File: src/extension/app/components/action-bar/menu-item/menu-item.css.js [hunk 2]
```diff
@@ -23,6 +23,17 @@ export const style = css`
     font-weight: 700;
   }
 
+  :host(.current-env.env-edit[aria-disabled="true"]) [name="description"]::slotted(*) {
+    font-weight: 400;
+    font-size: var(--spectrum-font-size-50);
+  }
+
+  :host(.current-env.env-edit[aria-disabled="true"]) {
+    background-color: var(--spectrum-gray-200);
```
Comment (dylandepass) [inline src/extension/app/components/action-bar/menu-item/menu-item.css.js:32]:
```suggestion
    background-color: var(--spectrum2-edit-background-default);
```

File: src/extension/app/components/plugin/env-switcher/env-switcher.js [hunk 3]
```diff
@@ -174,6 +174,21 @@ export class EnvironmentSwitcher extends ConnectedElement {
         },
       });
 
+      const currentEnvironment = this.currentEnv;
+      const descriptionText = currentEnvironment === 'edit'
```
Comment (dylandepass) [inline src/extension/app/components/plugin/env-switcher/env-switcher.js:178]: Could skip the re-assignment and just use `this.currentEnv`

File: test/app/components/plugin/plugin-action-bar.test.js [hunk 4]
```diff
@@ -237,7 +237,50 @@ describe('Plugin action bar', () => {
...
+    it('Editor - should display "last edited" description for Source', async () => {
+      sidekickTest
+        .mockFetchEditorStatusSuccess(HelixMockContentSources.SHAREPOINT, HelixMockContentType.DOC)
+        .mockFetchSidekickConfigSuccess(false, false)
+        .mockEditorAdminEnvironment(EditorMockEnvironments.EDITOR)
+        .createSidekick();
+
+      await sidekickTest.awaitStatusFetched();
+
+      await sidekickTest.awaitEnvSwitcher();
```
Comment (dylandepass) [inline test/app/components/plugin/plugin-action-bar.test.js:252]: Could just `awaitEnvSwitcher` if you don't care about the result of `awaitStatusFetched`

**Review by dylandepass — CHANGES_REQUESTED** [review]
Very nice @asthabh23 👏🏼

Let's also vertically center the icon
Screenshot 2024-10-07 at 9 23 06 AM

**Review by dylandepass — APPROVED** [review]

**Pull request merged**